bin/reth/src/cli/db_type.rs

```diff
@@ -0,0 +1,119 @@
+//! A real or test database type
+
+use crate::dirs::{ChainPath, DataDirPath, MaybePlatformPath};
+use reth_db::{
+    init_db,
+    test_utils::{create_test_rw_db, TempDatabase},
+    DatabaseEnv,
+};
+use reth_interfaces::db::LogLevel;
+use reth_primitives::Chain;
+use std::{str::FromStr, sync::Arc};
+
+/// A type that represents either a _real_ (represented by a path), or _test_ database, which will
+/// use a [TempDatabase].
+#[derive(Debug)]
+pub enum DatabaseBuilder {
+    /// The real database type, with a specified data dir
+    Real(MaybePlatformPath<DataDirPath>),
+    /// The test database type
+    Test,
+}
+
+impl DatabaseBuilder {
+    /// Creates a _test_ database
+    pub fn test() -> Self {
+        Self::Test
+    }
+
+    /// Initializes and returns the [DatabaseInstance] depending on the current database type. If
+    /// the [DatabaseBuilder] is test, the [LogLevel] is not used.
+    ///
+    /// If the [DatabaseBuilder] is test, then the [ChainPath] constructed will be derived from the
+    /// db path of the [TempDatabase] and the given chain.
+    pub fn build_db(
```

Comment:
can we rename this to `init_db` or something that makes it clear that this initializes the DB?

Reply:
yep, renamed and added more docs